### PR TITLE
Explicit websocket handlers

### PR DESCRIPTION
**Pull request by achamayou**
Extend http_method to RESTVerb (alternative naming suggestions welcome), so it's possible to explicitly register WS handlers. Context stays the same for now, we can fork it without user changes in the future if we decide it's useful to do so, but I'm not convinced right now.

Next step for #629

### CODE REVIEW

File: src/node/rpc/common_endpoint_registry.h [hunk 1]
```diff
@@ -247,7 +249,7 @@ namespace ccf
 
         for (auto& [verb, endpoint] : it->second)
         {
-          std::string verb_name = http_method_str(verb);
+          std::string verb_name = verb.c_str();
```
Comment (ashamis) [inline src/node/rpc/common_endpoint_registry.h:252]: is this right? I am assuming that verb is a std::string?

**Comment by ashamis** [comment]
This might be unrelated (or noise) but the websocket perf tests look like they may have regressed. Maybe run the pref tests a couple of times?

File: src/node/rpc/common_endpoint_registry.h (hunk 1 above)
Comment (achamayou) [reply src/node/rpc/common_endpoint_registry.h:252]: Verb is an RPCVerb now.

Comment (achamayou) [reply src/node/rpc/common_endpoint_registry.h:252]: Here we allocate a copy because we lowercase it for the schema.

**Comment by achamayou** [comment]
@ashamis I think that was just noise, there doesn't seem to be a noticeable difference.

**Pull request merged**